routing/endpointregistry_test.go

```diff
@@ -101,6 +104,38 @@ func TestDoRemovesOldEntries(t *testing.T) {
 	assert.Equal(t, int64(0), mRemoved.InflightRequests())
 }
 
+func TestRaceReadWrite(t *testing.T) {
+	r := routing.NewEndpointRegistry(routing.RegistryOptions{})
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		r.GetMetrics("some key")
+	}()
+	go func() {
+		defer wg.Done()
+		r.GetMetrics("some key").IncInflightRequest()
+	}()
+	wg.Wait()
+}
```

Comment:
I think they should do it in the loop during some time.

Reply:
I found no good sources to find out whether loop are required to detect races (I was able to detected races without loops in this particular case), so I will add such loop to be on the safe side.